SU-HSDO/suhumsci
# STN-293 : Color Pairing Architecture

# [STN-293](https://sparkbox.atlassian.net/browse/STN-293) READY FOR REVIEW

## Summary
Define the color pairings architecture and create a sample implementation using `src/scss/components/_pattern.hero.scss` in the colorful theme.

_NOTE:_ Color styles for the hero button and accent bar above the heading will be addressed in [STN-202](https://sparkbox.atlassian.net/browse/STN-202)

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. In the CLI, run `npm run test` to confirm all tests pass.
2. Go to http://economics.suhumsci.loc
3. To test the color pairing architecture sample implementation, you will need to find a page with the hero implemented. If you have pulled data from the dev site, you can find the hero on the QA page. http://economics.suhumsci.loc/qa/qa-page
4. The page should display the `ocean` color pairing by default.
5. Go to `Appearance` > `Settings` > `HumSci Colorful` and change the `Color Pairing` setting to ` Mountain`. View the page and confirm the color blocks in the hero have updated to the `Mountain` color pairing.
6. Update the color pairing setting to `Sand`. View the page and confirm the color blocks in the hero have updated to the `Sand` color pairing.

### Expected results
#### Ocean (default)
<img width="1039" alt="default" src="https://user-images.githubusercontent.com/12678977/77800359-ffd20200-704c-11ea-8f04-cfe9fd8fced5.png">

#### Mountain
<img width="1047" alt="mountain" src="https://user-images.githubusercontent.com/12678977/77800412-23954800-704d-11ea-8bcc-402f544f2059.png">

#### Sand
<img width="1025" alt="sand" src="https://user-images.githubusercontent.com/12678977/77800465-3c9df900-704d-11ea-87e1-6f3e05706cbd.png">

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
